### DISCUSSION TITLE
What is equality in JavaScript ?

### DISCUSSION THREAD

**Discussion by Dipali2377** [Dipali2377/Age-Calculator#2] [Q&A] [answered]
What is equality in JavaScript ?

**Comment by Akash2377** [comment, marked as answer]
JavaScript has both strict and type–converting comparisons:

Strict comparison (e.g., ===) checks for value equality without allowing coercion

Abstract comparison (e.g. ==) checks for value equality with coercion allowed
Some simple equality rules:

If either value (aka side) in comparison could be the true or false value, avoid == and use ===.

If either value in comparison could be of these specific values (0, "", or [] -- empty array), avoid == and use ===.

In all other cases, you're safe to use ==. Not only is it safe, but in many cases, it simplifies your code in a way that improves readability.
